p2p/net/swarm/swarm_listen.go

```diff
@@ -43,7 +43,7 @@ func (s *Swarm) ListenClose(addrs ...ma.Multiaddr) {
 
 	s.listeners.Lock()
 	for l := range s.listeners.m {
-		if !containsMultiaddr(addrs, l.Multiaddr()) {
+		if !containsSomeMultiaddr(addrs, l.Multiaddrs()) {
```

Comment:
We should extend the documentation for this function that we close a listener listening on multiple addresses if _any_ address matches.